test/unit/MemberRoles/stateManagement.js

```diff
@@ -0,0 +1,109 @@
+const { Role } = require('../utils').constants;
+const { expect } = require('chai');
+const { ethers } = require('hardhat');
+const { AddressZero } = ethers.constants;
+
+describe('state management', function () {
+  let membersCount;
+  before(function () {
+    const { members } = this.accounts;
+    membersCount = members.length + 1; // additional AB member
+  });
+
+  it('should return all members', async function () {
+    const { memberRoles } = this.contracts;
+    const {
+      members,
+      advisoryBoardMembers: [abMember],
+    } = this.accounts;
+
+    const { memberArray } = await memberRoles.members(Role.Member);
+    expect(memberArray.length).to.be.equal(membersCount);
+    members.forEach(member => {
+      expect(memberArray).to.include(member.address);
+    });
+    expect(memberArray).to.include(abMember.address);
+  });
+
+  it('should return all roles for a member', async function () {
+    const { memberRoles } = this.contracts;
+    const [member] = this.accounts.members;
+
+    const assignedRolesMember = await memberRoles.roles(member.address);
+    expect(assignedRolesMember[0]).to.be.equal(Role.Member);
+  });
+
+  it('should return authorized address for role', async function () {
+    const { memberRoles } = this.contracts;
+
+    const authorizedAddress = await memberRoles.authorized(Role.Member);
+    expect(authorizedAddress).to.be.equal(AddressZero);
+  });
+
+  it('should return length of all roles', async function () {
+    const { memberRoles } = this.contracts;
+
+    const [unAssignedMembersLength, advisoryBoardMembersLength, membersMembersLength] =
+      await memberRoles.getMemberLengthForAllRoles();
+    expect(unAssignedMembersLength).to.be.equal(0);
+    expect(advisoryBoardMembersLength).to.be.equal(1);
+    expect(membersMembersLength).to.be.equal(membersCount);
+  });
+
+  it('should return members length', async function () {
+    const { memberRoles } = this.contracts;
+
+    const membersLength = await memberRoles.membersLength(Role.Member);
+    expect(membersLength).to.be.equal(membersCount);
+  });
+
+  it('should return member at index', async function () {
+    const { memberRoles } = this.contracts;
+    const {
+      members: [member],
+    } = this.accounts;
+
+    const [memberAddress, isActive] = await memberRoles.memberAtIndex(Role.Member, 0);
+    expect(member.address).to.be.equal(memberAddress);
```

Comment:
this is the other way around.

expect(memberAddress).to.be.equal(member.address)